lega/utils/key.py

```diff
@@ -0,0 +1,159 @@
+import logging
+
+LOG = logging.getLogger(__name__)
+
+class Key():
+
+    def __init__(self, config):
+        """
+        Initiliazing the Key object from a dict
+        """
+        self.config = config
+
+    def public(self):
+        """
+        The public key part
+        Return bytes
+        """
+        raise NotImplementedError('subclasses of Key must provide an public() method')
+
+    def private(self):
+        """
+        The private key part
+        Return bytes
+        """
+        raise NotImplementedError('subclasses of Key must provide an private() method')
+
+
+
+class C4GHFileKey():
+    '''Loading a Crypt4GH-formatted file, and unlocking it using a passphrase.
+
+    See https://crypt4gh.readthedocs.io/en/latest/keys.html'''
+
+    def __init__(self, config):
+        
+        filepath = config.get('filepath')
+        passphrase = config.get('passphrase')
+        assert(filepath and passphrase)
+
+        # We use the parser provided in the crypt4gh.keys module
+        from crypt4gh.keys import get_private_key
+        LOG.debug('Unlocking private key: %s', filepath)
+        self.seckey = get_private_key(filepath, lambda : passphrase) # callback
+
+        # Don't bother reading the public key from file,
+        # We can simply recompute it from the private key
+        from nacl.public import PrivateKey
+        self.pubkey = bytes(PrivateKey(self.seckey).public_key)
+        LOG.debug('Setting public key: %s', self.pubkey.hex())
+
+        LOG.info('Successfully loaded a Crypt4GH-formatted key from file')        
+
+    def public(self):
+        return self.pubkey
+
+    def private(self):
+        return self.seckey
+
+
+class HashiCorpVaultKey():
+    pass
+
+
+class HTTPSKey():
```

Comment:
Inherit from baseclass
```suggestion
class HTTPSKey(Key):
```

Reply:
Note. I tried to copy the old content into that class, but I reeeeallly did not bother checking if I got it correctly. Since the java keyserver was failing when I deployed the master branch, I think it's up to the ones deploying the solution with a keyserver to implement the relevant class. Think of if as boilerplate code, to help them start.